no-case-declarations returns duplicate results

**Tell us about your environment**

* **ESLint Version:** v4.18.2
* **Node Version:** v9.7.1
* **npm Version:** 5.7.1

**What parser (default, Babel-ESLint, etc.) are you using?**

`default`

**Please show your full configuration:**

Configuration:
```json
{
  "parserOptions": {
    "sourceType": "module"
  },
  "rules": {
    "no-case-declarations": "error"
  }
}
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

<!-- Paste the source code below: -->
```js
export default function myFunction() {
  switch (true) {
    case true:
      const x = 1
      const y = 2
      return { x, y }
    default:
      return null
  }
}
```

<!-- Paste the command you used to run ESLint: -->
```bash
npx eslint .\foo.js
```

You can download the exact files from [this Gist](https://gist.github.com/Arcanemagus/8e7877a6d085f0e9cf9fb286ffc2435d).

**What did you expect to happen?**

A single error message is returned.

**What actually happened? Please include the actual, raw output from ESLint.**
CLI output
```
C:\temp\github\linter-ui-default_GH465\node_modules\eslint\bin\eslint.js

C:\temp\github\linter-ui-default_GH465\foo.js
  3:5  error  Unexpected lexical declaration in case block  no-case-declarations
  3:5  error  Unexpected lexical declaration in case block  no-case-declarations

✖ 2 problems (2 errors, 0 warnings)
```

Note that this isn't just a bug in the default CLI output, as the raw `report` from the `CLIEngine` shows the entire result object is identical:
```
> JSON.stringify(report.results.messages[0]) === JSON.stringify(report.results.messages[1])
true
```

Originally reported as https://github.com/steelbrain/linter-ui-default/issues/465, apologies if this has already been reported as I didn't find it in a search.

--- a/lib/rules/no-case-declarations.js
+++ b/lib/rules/no-case-declarations.js
@@ -50,7 +50,7 @@ module.exports = {
 
                     if (isLexicalDeclaration(statement)) {
                         context.report({
-                            node,
+                            node: statement,
                             messageId: "unexpected"
                         });
                     }

--- a/tests/lib/rules/no-case-declarations.js
+++ b/tests/lib/rules/no-case-declarations.js
@@ -41,42 +41,42 @@ ruleTester.run("no-case-declarations", rule, {
         {
             code: "switch (a) { case 1: let x = 1; break; }",
             parserOptions: { ecmaVersion: 6 },
-            errors: [{ messageId: "unexpected" }]
+            errors: [{ messageId: "unexpected", type: "VariableDeclaration" }]
         },
         {
             code: "switch (a) { default: let x = 2; break; }",
             parserOptions: { ecmaVersion: 6 },
-            errors: [{ messageId: "unexpected" }]
+            errors: [{ messageId: "unexpected", type: "VariableDeclaration" }]
         },
         {
             code: "switch (a) { case 1: const x = 1; break; }",
             parserOptions: { ecmaVersion: 6 },
-            errors: [{ messageId: "unexpected" }]
+            errors: [{ messageId: "unexpected", type: "VariableDeclaration" }]
         },
         {
             code: "switch (a) { default: const x = 2; break; }",
             parserOptions: { ecmaVersion: 6 },
-            errors: [{ messageId: "unexpected" }]
+            errors: [{ messageId: "unexpected", type: "VariableDeclaration" }]
         },
         {
             code: "switch (a) { case 1: function f() {} break; }",
             parserOptions: { ecmaVersion: 6 },
-            errors: [{ messageId: "unexpected" }]
+            errors: [{ messageId: "unexpected", type: "FunctionDeclaration" }]
         },
         {
             code: "switch (a) { default: function f() {} break; }",
             parserOptions: { ecmaVersion: 6 },
-            errors: [{ messageId: "unexpected" }]
+            errors: [{ messageId: "unexpected", type: "FunctionDeclaration" }]
         },
         {
             code: "switch (a) { case 1: class C {} break; }",
             parserOptions: { ecmaVersion: 6 },
-            errors: [{ messageId: "unexpected" }]
+            errors: [{ messageId: "unexpected", type: "ClassDeclaration" }]
         },
         {
             code: "switch (a) { default: class C {} break; }",
             parserOptions: { ecmaVersion: 6 },
-            errors: [{ messageId: "unexpected" }]
+            errors: [{ messageId: "unexpected", type: "ClassDeclaration" }]
         }
     ]
 });

